<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1044</width>
    <height>845</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>盲水印</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_4" columnstretch="2,1">
        <item row="7" column="1">
         <widget class="QFrame" name="frame_3">
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QGridLayout" name="gridLayout">
           <item row="0" column="2">
            <widget class="QSpinBox" name="spinBox_2">
             <property name="maximum">
              <number>2147483647</number>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_2">
             <property name="toolTip">
              <string>推荐只填第一个, 数值大约为32左右, 在嵌入后图片与原图相似度可接受的前提下尽量取大</string>
             </property>
             <property name="text">
              <string>量化因子</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QDoubleSpinBox" name="doubleSpinBox">
             <property name="maximum">
              <double>399.990000000000009</double>
             </property>
             <property name="value">
              <double>1.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="2" column="2">
            <widget class="QDoubleSpinBox" name="doubleSpinBox_2">
             <property name="maximum">
              <double>399.990000000000009</double>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="label">
             <property name="toolTip">
              <string>两个随机的整数即可</string>
             </property>
             <property name="text">
              <string>随机种子</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QSpinBox" name="spinBox_3">
             <property name="minimum">
              <number>0</number>
             </property>
             <property name="maximum">
              <number>2147483647</number>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="5" column="1">
         <widget class="QFrame" name="frame_2">
          <property name="toolTip">
           <string>从剪切板导入密钥, 同时修改下面的参数. 注, 点击左侧栏中的密钥可以将密钥放到剪切板中</string>
          </property>
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QGridLayout" name="gridLayout_2" columnstretch="1,1">
           <property name="sizeConstraint">
            <enum>QLayout::SetDefaultConstraint</enum>
           </property>
           <item row="2" column="0">
            <widget class="QPushButton" name="pushButton_2">
             <property name="text">
              <string>导入水印</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string/>
             </property>
             <property name="scaledContents">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string/>
             </property>
             <property name="wordWrap">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QPushButton" name="pushButton">
             <property name="text">
              <string>导入图片</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QFrame" name="frame">
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QGridLayout" name="gridLayout_6">
           <item row="0" column="1">
            <widget class="QRadioButton" name="radioButton">
             <property name="toolTip">
              <string>提取模式 , 只需在 导入图片 导入嵌入水印后的图</string>
             </property>
             <property name="text">
              <string>提取</string>
             </property>
             <property name="checked">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QRadioButton" name="radioButton_2">
             <property name="toolTip">
              <string>嵌入模式</string>
             </property>
             <property name="text">
              <string>嵌入</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <widget class="QCheckBox" name="checkBox_2">
             <property name="toolTip">
              <string>勾选后会在同时输出嵌入水印后的图与原图的相识的参数</string>
             </property>
             <property name="text">
              <string>同时进行相似度验证</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="12" column="1">
         <widget class="QFrame" name="frame_5">
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QGridLayout" name="gridLayout_5">
           <item row="0" column="0">
            <widget class="QPushButton" name="pushButton_3">
             <property name="text">
              <string>嵌入</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QPushButton" name="pushButton_4">
             <property name="statusTip">
              <string>在设置工作目录的前提下, 将原图和水印图复制到工作目录下</string>
             </property>
             <property name="text">
              <string>将图片复制到工作目录</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QPushButton" name="pushButton_11">
             <property name="text">
              <string>打开工作目录</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QFrame" name="frame_6">
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QGridLayout" name="gridLayout_7">
           <item row="0" column="0">
            <widget class="QPushButton" name="pushButton_5">
             <property name="toolTip">
              <string>设置工作目录后,会在嵌入时在工作目录写入密钥: key.json</string>
             </property>
             <property name="text">
              <string>设置工作目录</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QPushButton" name="pushButton_6">
             <property name="text">
              <string>从剪切板导入密钥</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QPushButton" name="pushButton_7">
             <property name="toolTip">
              <string>从key.json文件导入密钥, 并更新下来参数</string>
             </property>
             <property name="text">
              <string>从文件导入密钥</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLabel" name="label_10">
             <property name="text">
              <string>未设置工作目录</string>
             </property>
             <property name="wordWrap">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="14" column="0">
         <widget class="QProgressBar" name="BlueProgressBar">
          <property name="styleSheet">
           <string notr="true">#BlueProgressBar {
    border: 2px solid #2196F3;/*边框以及边框颜色*/
    border-radius: 5px;
    background-color: #E0E0E0;
}
#BlueProgressBar::chunk {
    background-color: #2196F3;
    width: 10px; /*区块宽度*/
    margin: 0.5px;
}</string>
          </property>
          <property name="value">
           <number>0</number>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
          <property name="invertedAppearance">
           <bool>false</bool>
          </property>
          <property name="format">
           <string>%p%</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0" rowspan="12">
         <widget class="QListView" name="listView"/>
        </item>
        <item row="9" column="1" rowspan="2">
         <widget class="QFrame" name="frame_4">
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QGridLayout" name="gridLayout_3">
           <item row="0" column="1">
            <widget class="QSpinBox" name="spinBox">
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>10</number>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>小波变换级数</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QCheckBox" name="checkBox">
             <property name="text">
              <string>限制相同</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_7">
             <property name="text">
              <string>水印形状</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_6">
             <property name="text">
              <string>分块形状</string>
             </property>
            </widget>
           </item>
           <item row="2" column="2">
            <widget class="QSpinBox" name="spinBox_5">
             <property name="minimum">
              <number>4</number>
             </property>
             <property name="maximum">
              <number>16</number>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QSpinBox" name="spinBox_4">
             <property name="minimum">
              <number>4</number>
             </property>
             <property name="maximum">
              <number>16</number>
             </property>
            </widget>
           </item>
           <item row="1" column="2">
            <widget class="QSpinBox" name="spinBox_7">
             <property name="maximum">
              <number>222222222</number>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QSpinBox" name="spinBox_6">
             <property name="maximum">
              <number>222222222</number>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="Line" name="line_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="11" column="1">
         <widget class="Line" name="line_4">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="Line" name="line">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="6" column="1">
         <widget class="Line" name="line_3">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="8" column="1">
         <widget class="Line" name="line_5">
          <property name="lineWidth">
           <number>1</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_4">
       <attribute name="title">
        <string>工具</string>
       </attribute>
       <widget class="QFrame" name="frame_7">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>280</width>
          <height>201</height>
         </rect>
        </property>
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Sunken</enum>
        </property>
        <layout class="QGridLayout" name="gridLayout_8">
         <item row="0" column="0" colspan="2">
          <widget class="QTextBrowser" name="textBrowser_2">
           <property name="html">
            <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'SimSun'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:11pt;&quot;&gt;恢复形变攻击&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;将经过平移,旋转,缩放,透视等形变的图片恢复成接近原图,是从受到形变攻击的图片中提取水印的必要前置步骤&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QPushButton" name="pushButton_8">
           <property name="text">
            <string>读取原图</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QPushButton" name="pushButton_10">
           <property name="text">
            <string>恢复</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QPushButton" name="pushButton_9">
           <property name="text">
            <string>读取受到攻击的图片</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QFrame" name="frame_8">
           <property name="frameShape">
            <enum>QFrame::StyledPanel</enum>
           </property>
           <property name="frameShadow">
            <enum>QFrame::Raised</enum>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <property name="leftMargin">
             <number>0</number>
            </property>
            <property name="topMargin">
             <number>0</number>
            </property>
            <property name="rightMargin">
             <number>0</number>
            </property>
            <property name="bottomMargin">
             <number>0</number>
            </property>
            <item>
             <widget class="QLabel" name="label_12">
              <property name="text">
               <string>阈值:</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="doubleSpinBox_3">
              <property name="maximum">
               <double>1.000000000000000</double>
              </property>
              <property name="singleStep">
               <double>0.100000000000000</double>
              </property>
              <property name="value">
               <double>0.700000000000000</double>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>盲水印抗破坏验证</string>
       </attribute>
       <widget class="QLabel" name="label_8">
        <property name="geometry">
         <rect>
          <x>200</x>
          <y>180</y>
          <width>72</width>
          <height>15</height>
         </rect>
        </property>
        <property name="text">
         <string>咕咕咕</string>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="tab_3">
       <attribute name="title">
        <string>关于</string>
       </attribute>
       <widget class="QGraphicsView" name="graphicsView">
        <property name="geometry">
         <rect>
          <x>400</x>
          <y>420</y>
          <width>256</width>
          <height>251</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">border-image: url(:/pic/setu_qrcode.png);</string>
        </property>
       </widget>
       <widget class="QGraphicsView" name="graphicsView_2">
        <property name="geometry">
         <rect>
          <x>660</x>
          <y>360</y>
          <width>256</width>
          <height>371</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">border-image: url(:/pic/alipay.png);</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_9">
        <property name="geometry">
         <rect>
          <x>410</x>
          <y>350</y>
          <width>121</width>
          <height>51</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <family>Adobe Arabic</family>
          <pointsize>14</pointsize>
         </font>
        </property>
        <property name="text">
         <string>捐赠通道</string>
        </property>
        <property name="scaledContents">
         <bool>false</bool>
        </property>
       </widget>
       <widget class="QTextBrowser" name="textBrowser">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>891</width>
          <height>361</height>
         </rect>
        </property>
        <property name="html">
         <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'SimSun'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:11pt;&quot;&gt;菜逼程序员不会写那种可以弹出的帮助和关于界面&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-size:11pt;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:11pt;&quot;&gt;帮助部分:&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:11pt;&quot;&gt;鼠标停在大多数元素上会有提示帮助的&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:11pt;&quot;&gt;左侧栏,密钥那一行点击可以把密钥放到剪切板上,图片那一栏点击可以查看图片&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:11pt;&quot;&gt;相似度参数解析:&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:11pt;&quot;&gt;psnr在40以上是效果极好,在30~40为好,一般增大量化因子使其在30~40,可以保证较好的抗攻击能力和与原图较小的失真度&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:11pt;&quot;&gt;至于NCC值,越高表示效果越好,没找到相关指标, 反正保证略大于0.99就行&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:11pt;&quot;&gt;至于小波级数和分块大小,都是大一点好,小波级数一般取1~3,分块大小一般取4*4或者8*8 , 但是需要满足以下条件&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:11pt;&quot;&gt;设小波变换级数为d默认1,分块大小为(m,m)默认(4,4),图片大小为x * y ,水印的长宽乘积为wm_size,需满足&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;img src=&quot;:/pic/gongshi.png&quot; /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
       <widget class="QGraphicsView" name="graphicsView_3">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>410</y>
          <width>256</width>
          <height>251</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">border-image: url(:/pic/1563978351.png);</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_11">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>370</y>
          <width>271</width>
          <height>41</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <family>Adobe Arabic</family>
          <pointsize>14</pointsize>
         </font>
        </property>
        <property name="text">
         <string>请star一下我的github项目</string>
        </property>
       </widget>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1044</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>文件</string>
    </property>
   </widget>
   <widget class="QMenu" name="menu_2">
    <property name="title">
     <string>关于</string>
    </property>
    <addaction name="help"/>
    <addaction name="support"/>
   </widget>
   <widget class="QMenu" name="menu_3">
    <property name="title">
     <string>设置</string>
    </property>
   </widget>
   <addaction name="menu"/>
   <addaction name="menu_3"/>
   <addaction name="menu_2"/>
  </widget>
  <action name="help">
   <property name="icon">
    <iconset resource="img/sources.qrc">
     <normaloff>:/icon/question-circle.png</normaloff>:/icon/question-circle.png</iconset>
   </property>
   <property name="text">
    <string>帮助</string>
   </property>
  </action>
  <action name="support">
   <property name="icon">
    <iconset>
     <normalon>:/icon/red envelope.png</normalon>
     <selectedon>:/icon/red envelope.png</selectedon>
    </iconset>
   </property>
   <property name="text">
    <string>支持</string>
   </property>
   <property name="toolTip">
    <string>恰饭</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="img/sources.qrc"/>
 </resources>
 <connections/>
</ui>
